### PR TITLE
fix: fix S3 Data Lake connector empty role_arn validation

### PR DESCRIPTION
**Pull request by davinchia**
- Change role_arn validation to check for both null and blank strings
- Add validation for mismatched AWS credentials (only one of access_key_id or secret_access_key provided)
- Add comprehensive tests for empty string role_arn, null role_arn, and credential validation
- Improve error messages for better debugging

Fixes airbytehq/airbyte-internal-issues#14643

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

> [!IMPORTANT]
> **Auto-merge enabled.**
> 
> _This PR is set to merge automatically when all requirements are met._

### CODE REVIEW

**Review by benmoriceau — APPROVED** [review]

File: airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/io/S3DataLakeUtil.kt [hunk 1]
```diff
@@ -273,6 +271,14 @@ class S3DataLakeUtil(
         // For example, this should allow us to use AWS instance profiles.
         val awsAccessKeyId = config.awsAccessKeyConfiguration.accessKeyId
         val awsSecretAccessKey = config.awsAccessKeyConfiguration.secretAccessKey
+
+        // Validate that either both credentials are provided or neither (for instance profiles)
+        if ((awsAccessKeyId == null) != (awsSecretAccessKey == null)) {
+            throw IllegalArgumentException(
+                "Both access_key_id and secret_access_key must be provided together, or neither (to use instance profiles)"
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/io/S3DataLakeUtil.kt]: in assume role mode, we always use the key+secret https://github.com/airbytehq/airbyte/blob/78c43755e47bc35c3ee64034f894c257993a0dbb/airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/GlueCredentialsProvider.kt#L58

**Review by edgao — APPROVED** [review]
one comment about logic, otherwise lgtm

**Comment by davinchia** [comment]
/format-fix
> Format-fix job started... [Check job output.][1]

[1]: https://github.com/airbytehq/airbyte/actions/runs/18199144356
> ✅ Changes applied successfully. (b4b2f453b0ce9e01e9200eeabbad9c3d2f4ade66)

File: airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/io/S3DataLakeUtil.kt (hunk 1 above)
Comment (davinchia) [reply airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/io/S3DataLakeUtil.kt]: Done!

PTAL @edgao

**Review by edgao — APPROVED** [review]
